src/basic/pidref.c

```diff
@@ -84,8 +84,8 @@ int pidref_set_pid(PidRef *pidref, pid_t pid) {
 
         fd = pidfd_open(pid, 0);
         if (fd < 0) {
-                /* Graceful fallback in case the kernel doesn't support pidfds or is out of fds */
-                if (!ERRNO_IS_NOT_SUPPORTED(errno) && !ERRNO_IS_PRIVILEGE(errno) && !ERRNO_IS_RESOURCE(errno))
+                /* Graceful fallback in case the kernel is out of fds */
+                if (!ERRNO_IS_RESOURCE(errno))
```

Comment:
note that this also means we cannot run anymore in container environments which block pidfd_open(). So far we fell back to pid-based operation in that case. Which i guess is a bit ugly, but if we change that I'd prefer if we changed that another time. ence please keep the ERRNO_IS_PRIVILEGED() thin in place.

(container envs tend to implement syscall allow lists, hence block all syscalls that are not known at the moment they are deployed, which likely covers pidfd_open() for many container enviroments currently in use)

Reply:
(we should probably add a NEWS entry that asks container mgrs to update their seccomp filters *NOW*, in order not to risk that systemd payloads will soon stop working.